board.py

```diff
@@ -215,11 +215,26 @@ def insertCard(self, inputArgs):
         if not self.cardLocationIsValidSpot(positionNewCard, positionSecondTile, newCard):
             print("The location where you want to place your card is not valid.")
             return None
-        self.board[positionNewCard[1]][positionNewCard[0]] = newCard.activeSide.tile1
-        self.board[positionSecondTile[1]][positionSecondTile[0]] = newCard.activeSide.tile2
-
-        self.nbrCards += 1
-
+        
```

Comment:
Please add a comment specifying what int(inputArgs[3]) == 1 means.
We know this refers to the possibility of placing cards on the first row.
Why are there only 4 rotation codes out of the 8 that are valid with a card on the first row?

Reply:
So the reason that only 4 rotation codes are specified is because we were having negative tile placement issues with the vertical cards, not the horizontal ones. So these are just edge cases for vertical cards places on the first row. The other rotation code cards (horizontal cards) work fine as is.

Reply:
i thought negative tile placement issues could occur with the top of the board too, or if a card is placed on a far right tile horizontally (so that the second tile to the right is out of bounds)